cli/cmd/config.go

```diff
@@ -22,66 +24,124 @@ var configCmd = &cobra.Command{
 var setConfigCmd = &cobra.Command{
 	Use:   "set <property> <value>",
 	Short: "Set a configuration property in Odigos",
-	Args:  cobra.ExactArgs(2),
+	Args:  cobra.MinimumNArgs(2),
 	Run: func(cmd *cobra.Command, args []string) {
 		property := args[0]
-		value := args[1]
+		value := args[1:]
 
 		ctx := cmd.Context()
 		client := cmdcontext.KubeClientFromContextOrExit(ctx)
 		ns, _ := cmd.Flags().GetString("namespace")
 		if ns == "" {
-			ns = "odigos-system"
+			ns = consts.DefaultOdigosNamespace
 		}
 
 		l := log.Print(fmt.Sprintf("Updating %s to %s...", property, value))
-		err := updateConfigProperty(ctx, client, ns, property, value)
+
+		config, err := resources.GetCurrentConfig(ctx, client, ns)
+		if err != nil {
+			odigosConfig, err := resources.GetDeprecatedConfig(ctx, client, ns)
+			if err != nil {
+				l.Error(fmt.Errorf("unable to read the current Odigos configuration: %w", err))
+				os.Exit(1)
+			}
+			config = odigosConfig.ToCommonConfig()
+		}
+
+		config.ConfigVersion += 1
+		err = setConfigProperty(config, property, value)
 		if err != nil {
 			l.Error(err)
 			os.Exit(1)
 		}
-		l.Success()
 
-		if property == "central-backend-url" {
-			l = log.Print("Restarting UI pod to apply changes...")
-			err = restartUIPod(ctx, client, ns)
-			if err != nil {
-				l.Error(err)
-				os.Exit(1)
-			}
-			l.Success()
+		currentTier, err := odigospro.GetCurrentOdigosTier(ctx, client, ns)
+		if err != nil {
+			l.Error(fmt.Errorf("unable to read the current Odigos tier: %w", err))
+			os.Exit(1)
+		}
+
+		resourceManagers := resources.CreateResourceManagers(client, ns, currentTier, nil, config, "Updating Config")
+		err = resources.ApplyResourceManagers(ctx, client, resourceManagers, "Updating Config")
+		if err != nil {
+			l.Error(fmt.Errorf("failed to apply updated configuration: %w", err))
+			os.Exit(1)
 		}
+
+		err = resources.DeleteOldOdigosSystemObjects(ctx, client, ns, config)
+		if err != nil {
+			fmt.Println("Odigos config update failed - unable to cleanup old Odigos resources.")
+			os.Exit(1)
+		}
+
+		l.Success()
+
 	},
 }
 
-func updateConfigProperty(ctx context.Context, client *kube.Client, ns, property, value string) error {
-	configMapName := "odigos-config"
+func setConfigProperty(config *common.OdigosConfiguration, property string, value []string) error {
```

Comment:
should we err in case the property expects 1 value but the `value` array has more than one?
instead of ignoring it